src/theory/output_channel.h

```diff
@@ -32,6 +32,36 @@
 namespace CVC4 {
 namespace theory {
 
+/** Properties of lemmas */
+enum class LemmaProperty : uint32_t
```

Comment:
Is there a particular reason why we explicitly specify the underlying datatype? (Why would we want to enforce a `uint32_t`?) I'd suggest to simply drop it.

```suggestion
enum class LemmaProperty
```

Reply:
I am actually using this in the explicit casting for `operator&`. Dropping this would not ensure the proper cast.
Alternatively, I noticed its possible to get the underlying type automatically via type traits using C++14? https://en.cppreference.com/w/cpp/header/type_traits
I think I'd rather just hard code this for now.

Reply:
According to https://en.cppreference.com/w/cpp/language/enum, the underlying type defaults to `int`. Unless you need way more bits, you could remove the explicit type specification and cast to `int`...

Reply:
I'm not sure what our coding guidelines say for this.  @aniemetz @4tXJ7f  ?

Reply:
I think that since we use bit-wise operators (`|`) on the values of the enum, we should explicitly specify an unsigned type. Yes, it doesn't really matter since the values are all within the range of positive integer values but applying bit-wise operators to signed integers still feels wrong.